library/core/src/slice/raw.rs

```diff
@@ -90,6 +90,7 @@ use crate::ptr;
 #[stable(feature = "rust1", since = "1.0.0")]
 #[rustc_const_stable(feature = "const_slice_from_raw_parts", since = "1.64.0")]
 #[must_use]
+#[rustc_diagnostic_item = "slice_from_raw_parts"]
```

Comment:
This is confusing with the function that is literally named `slice_from_raw_parts` (in `core::ptr`), not sure what the best name would be for this diag item.

Reply:
This one's unfortunate. There isn't really a better way to name. It does help that basically all free functions are named `module_name_function_name` and that does disambiguate.